### DISCUSSION TITLE
misspelling of the location names

### DISCUSSION THREAD

**Discussion by Atefeh197** [openai/whisper#2260] [General]
Hi 

Is there any way to add a list of location names to Whisper's decoder, now it generates them with misspellings!
I know about the prompt but it gets only 200 words!

**Comment by glangford** [comment]
Whisper does not have a method for adding special names beyond prompting.

One approach would be to post-process the transcript to correct misspellings. See for example

Python: How to correct misspelled names
https://stackoverflow.com/questions/41192424/python-how-to-correct-misspelled-names

Jellyfish library
https://github.com/jamesturk/jellyfish

**Reply by Atefeh197** [reply]
thank you,

These links are appropriate for a list of words, I have a text, It is possible to change many correct words! for example I have a location "Whyte Avenue", it is similar to "white"! 
I do not know if I can fine-tune only the location's name without any context. For example provide speech and text of "Whyte Avenue"!

**Reply by glangford** [reply]
> I do not know if I can fine-tune only the location's name without any context.

You can include context as appropriate. Using `difflib` for example,

```
>>> from difflib import get_close_matches
>>> get_close_matches('white avenue', ['New York', 'Brown Street', 'Whyte Avenue'])
['Whyte Avenue']
>>> get_close_matches('wait avenue', ['New York', 'Brown Street', 'Whyte Avenue'])
['Whyte Avenue']
```

But depending on the transcription and the place name, you could get false matches of course. For example the audio might accurately transcribe to "I walked down a wide avenue" which sounds similar to "Whyte Avenue". This can be tuned, and requires some experimentation eg.

```
>>> get_close_matches('wide avenue', ['New York', 'Brown Street', 'Whyte Avenue'])
['Whyte Avenue']
>>> get_close_matches('wide avenue', ['New York', 'Brown Street', 'Whyte Avenue'], cutoff=0.7)
[]
```

And there is still additional work needed because of the context - since we are matching 2 words at a time in this case!

So building up this toy example a bit more, 

```
>>> from itertools import pairwise
>>> sentence = "The truck stopped on white avenue."
>>> locations = ['New York', 'Brown Street', 'Whyte Avenue']
>>> [pair for pair in pairwise(sentence.split()) if get_close_matches(" ".join(pair), locations)]
[('white', 'avenue.')]
```

Anyway hopefully these examples are helpful. You could also google 'python regex fuzzy match' for another possible approach. I don't use either of these myself so I'm not sure what might be best for your use case.
